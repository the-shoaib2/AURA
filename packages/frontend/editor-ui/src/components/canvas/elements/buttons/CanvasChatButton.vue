<script lang="ts" setup>
import { type ButtonProps } from '@aura/design-system';

export type Props = Pick<ButtonProps, 'label' | 'type'>;

defineProps<Props>();
</script>
<template>
	<auraButton
		:label="label"
		size="large"
		icon="message-circle"
		:type="type"
		data-test-id="workflow-chat-button"
	/>
</template>
