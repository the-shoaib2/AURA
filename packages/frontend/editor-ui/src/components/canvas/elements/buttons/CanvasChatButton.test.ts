import { createComponentRenderer } from '@/__tests__/render';
import CanvasChatButton from './CanvasChatButton.vue';

const renderComponent = createComponentRenderer(CanvasChatButton, {
	global: {
		stubs: {
			auraButton: true,
		},
	},
});

describe('CanvasChatButton', () => {
	it('should render correctly', () => {
		const wrapper = renderComponent();

		expect(wrapper.html()).toMatchSnapshot();
	});
});
