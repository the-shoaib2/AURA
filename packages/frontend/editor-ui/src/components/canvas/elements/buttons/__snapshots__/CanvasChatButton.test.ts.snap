// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`CanvasChatButton > should render correctly 1`] = `"<aura-button-stub icon="message-circle" block="false" element="button" label="" square="false" active="false" disabled="false" loading="false" outline="false" size="large" text="false" type="primary" data-test-id="workflow-chat-button"></aura-button-stub>"`;
